sdk-extensions/jaeger-remote-sampler/src/main/java/io/opentelemetry/sdk/extension/trace/jaeger/sampler/DefaultGrpcServiceBuilder.java

```diff
@@ -0,0 +1,167 @@
+  private long timeoutNanos;
+  private URI endpoint;
+  private boolean compressionEnabled = false;
+  @Nullable private Metadata metadata;
+  @Nullable private byte[] trustedCertificatesPem;
+  @Nullable private RetryPolicy retryPolicy;
+
+  /** Creates a new {@link OkHttpGrpcExporterBuilder}. */
+  // Visible for testing
+  DefaultGrpcServiceBuilder(
+      String type,
+      Function<ManagedChannel, MarshalerServiceStub<ReqT, ResT, ?>> stubFactory,
+      long defaultTimeoutSecs,
+      URI defaultEndpoint,
+      String grpcServiceName) {
+    this.type = type;
+    this.stubFactory = stubFactory;
+    this.grpcServiceName = grpcServiceName;
+    timeoutNanos = TimeUnit.SECONDS.toNanos(defaultTimeoutSecs);
+    endpoint = defaultEndpoint;
+  }
+
+  @Override
+  public DefaultGrpcServiceBuilder<ReqT, ResT> setChannel(ManagedChannel channel) {
+    requireNonNull(channel, "channel");
+    this.channel = channel;
+    return this;
+  }
+
+  @Override
+  public DefaultGrpcServiceBuilder<ReqT, ResT> setTimeout(long timeout, TimeUnit unit) {
+    requireNonNull(unit, "unit");
+    checkArgument(timeout >= 0, "timeout must be non-negative");
+    timeoutNanos = unit.toNanos(timeout);
+    return this;
+  }
+
+  @Override
+  public DefaultGrpcServiceBuilder<ReqT, ResT> setTimeout(Duration timeout) {
+    requireNonNull(timeout, "timeout");
+    checkArgument(!timeout.isNegative(), "timeout must be non-negative");
+    return setTimeout(timeout.toNanos(), TimeUnit.NANOSECONDS);
+  }
+
+  @Override
+  public DefaultGrpcServiceBuilder<ReqT, ResT> setEndpoint(String endpoint) {
+    requireNonNull(endpoint, "endpoint");
+    this.endpoint = ExporterBuilderUtil.validateEndpoint(endpoint);
+    return this;
+  }
+
+  @Override
+  public DefaultGrpcServiceBuilder<ReqT, ResT> setCompression(String compressionMethod) {
+    requireNonNull(compressionMethod, "compressionMethod");
+    checkArgument(
+        compressionMethod.equals("gzip") || compressionMethod.equals("none"),
+        "Unsupported compression method. Supported compression methods include: gzip, none.");
+    this.compressionEnabled = true;
+    return this;
+  }
+
+  @Override
+  public DefaultGrpcServiceBuilder<ReqT, ResT> setTrustedCertificates(
+      byte[] trustedCertificatesPem) {
+    requireNonNull(trustedCertificatesPem, "trustedCertificatesPem");
+    this.trustedCertificatesPem = trustedCertificatesPem;
+    return this;
+  }
+
+  @Override
+  public DefaultGrpcServiceBuilder<ReqT, ResT> addHeader(String key, String value) {
+    requireNonNull(key, "key");
+    requireNonNull(value, "value");
+    if (metadata == null) {
+      metadata = new Metadata();
+    }
+    metadata.put(Metadata.Key.of(key, ASCII_STRING_MARSHALLER), value);
+    return this;
+  }
+
+  @Override
+  public DefaultGrpcServiceBuilder<ReqT, ResT> addRetryPolicy(RetryPolicy retryPolicy) {
+    requireNonNull(retryPolicy, "retryPolicy");
+    this.retryPolicy = retryPolicy;
+    return this;
+  }
```

Comment:
I think in a soon followup we can add all except `addRetryPolicy` to the builder. I don't think it hurts too much keeping them.

`setChannel` we hope is not used but since it was public let's stick with having it for now and deprecating all of our `setChannel` methods in one PR